### ISSUE TITLE
Allow rules to be fired in the pageAPI

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#17274]
Customers want to be able to fire rules when calling the pageAPI, if the page had rules defined in to. We need to add an optional param called fireRules that defaults to false that will be fired if the param is passed as true

**Comment by wezell** [comment]
Rules are fired when called via API. In the response, we now include the Visitor object, if it is set.

```json
  "viewAs" : {
    "visitor" : {
      "tags" : [ ],
      "device" : "UNKNOWN",
      "isNew" : true,
      "userAgent" : {
        "operatingSystem" : "UNKNOWN",
        "browser" : "DOWNLOAD",
        "id" : 16843024,
        "browserVersion" : null
      },
      "dmid" : "404f38e9-05e9-4dc1-bbe2-357381e09d20",
      "personas" : [ ]
    },
    "language" : {
      "id" : 1,
      "languageCode" : "en",
      "countryCode" : "US",
      "language" : "English",
      "country" : "United States"
    },
    "mode" : "LIVE"
  }
```

**Comment by wezell** [comment]
Additionally, implicit personalization has changed.  While a Visitor still has a persona, we also keep a list of all the personas that the visitor has been assigned.  You can see this list by calling $visitor.weightedPersonas - returns a map of the persona and the percentage of times that persona  has been assigned to the visitor and the percentage of how often that persona has been set e.g. if someone has been classified as  extremeSports 4 times, a snowboarder 3 times and a skiier 1 time, this would return:
```
{
   extremeSports:.5,
   snowboarder:.375,
   skiier:.125
 }
```

Personalization takes these percentages into account when calculating what content to return.  This behavior can be disabled by setting `PULLPERSONALIZED_USE_MULTIPLE_PERSONAS=false` in the dotmarketing-config.properties.

By default all personas are weighted the same, meaning, if someone has been personified as "skier" and then later as "surfer", then these two personas would have equal weight in pulling personalized content.  If you would like to weight the later persona higher than the previous personas, you can set an additional weight to give the last persona by setting this in your config: `PULLPERSONALIZED_LAST_PERSONA_WEIGHT=(int, 0-100)`

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/17305

**Comment by jdotcms** [comment]
I have done personalization test with rules.
I have set a content X for default
content Y for Retiree and content Z for Global

Then I have set the rules:
if request parameter 'p' is 'r', set Retiree as a visitor.
if request parameter 'p' is 'z;, set Global as a visitor.

Then I open a new browser and request the page without the 'p' parameter, the content returned is X which is ok.

Then, I request the page with the following query string: ?p=r 
In order to get the Retiree persona set and retrieve the content Y, the personalization works fine and still working for ?p=g (which is the Z content for global)
The issue is on the default page, the default keeps the latest persona set, so if you open default page, you get content Y which is the content associated to Retiree instead of X which should be the content for default (no query string)
